flake.nix

```diff
@@ -5,33 +5,99 @@
+    pre-commit-hooks-nix.url = "github:cachix/pre-commit-hooks.nix";
   };
 
   outputs = inputs@{ flake-parts, rust-overlay, ... }:
     flake-parts.lib.mkFlake { inherit inputs; } {
-      imports = [ ];
+      imports = [
+        inputs.pre-commit-hooks-nix.flakeModule
+      ];
 
       systems = [ "x86_64-linux" "aarch64-linux" "aarch64-darwin" "x86_64-darwin" ];
 
       perSystem = { config, self', inputs', pkgs, system, ... }:
         let
           rustPkgs = pkgs.appendOverlays [ (import rust-overlay) ];
           rust = rustPkgs.rust-bin.fromRustupToolchainFile ./web/rust-toolchain.toml;
+          goPkgs = with pkgs; [
+            go
+            gotools
+          ];
         in
         {
+          pre-commit = {
+            check.enable = true;
+            pkgs = pkgs;
+            settings = {
+              hooks = {
+                gofmt.enable = true;
+                govet.enable = true;
+                golangci-lint = {
+                  enable = true;
+                  name = "golangci-lint";
+                  description = "Lint my golang code";
+                  files = "\.go$";
+                  entry =
+                    let
+                      script = pkgs.writeShellScript "precommit-golangci" ''
+                        set -e
+                        export PATH=$PATH:${pkgs.go}/bin
+                        ${pkgs.golangci-lint}/bin/golangci-lint run --new-from-rev HEAD --fix
+                      '';
+                    in
+                    builtins.toString script;
+                  require_serial = true;
+                  pass_filenames = false;
+                };
+                goimports = {
+                  enable = true;
+                  name = "goimports";
+                  description = "Format my golang code";
+                  files = "\.go$";
+                  entry =
+                    let
+                      script = pkgs.writeShellScript "precommit-goimports" ''
+                        set -e
+                        failed=false
+                        for file in "$@"; do
+                            # redirect stderr so that violations and summaries are properly interleaved.
+                            if ! ${pkgs.gotools}/bin/goimports -l -d "$file" 2>&1
+                            then
+                                failed=true
+                            fi
+                        done
+                        if [[ $failed == "true" ]]; then
+                            exit 1
+                        fi
+                      '';
+                    in
+                    builtins.toString script;
+                };
+              };
+            };
+          };
+
           formatter = pkgs.nixpkgs-fmt;
 
           devShells.default = pkgs.mkShell {
             PLAYWRIGHT_BROWSERS_PATH = pkgs.playwright-driver.browsers;
+            buildInputs = config.pre-commit.settings.enabledPackages;
```

Comment:
Doesn't this override the `devShells.default.packages` attribute?

Reply:
You are right, wrong place.

I am moving `enabledPackages` under `packages` and everything works as expected..

```sh
$ which golangci-lint
/nix/store/5pjiagw5zn3xfws2wdnaw094p88n6jaa-golangci-lint-1.57.2/bin/golangci-lint
$ nix flake check && echo $?
0
```